src/index.jsx

```diff
@@ -25,6 +26,15 @@ const App = () => (
   </Provider>
 );
 
+const Unauthorized = () => "You are not authorized to access the gradebook for this course.";
+
 if (apiClient.ensurePublicOrAuthencationAndCookies(window.location.pathname)) {
-  ReactDOM.render(<App />, document.getElementById('root'));
+  LmsApiService.canUserViewGradebook(window.location.pathname.substring(1)).then(
```

Comment:
Could you please give an variable to `window.location.pathname.substring(1)`? I don't really know what this substring represent. Also, please handle the case if the `substring(1)` is empty or null

Reply:
this is the course key. it includes the leading "/", which the substring removes.
You're right, I should be checking if it's empty